File: QID-1482-MVAtdistail/MVAtdistail.r

# clear variables and close windows
rm(list = ls(all = TRUE))
graphics.off()

# Tail comparison of t-distribution
par(mar = c(5, 5, 5, 5))
par(mfrow = c(1, 1))
xx = seq(-6, 6, by = 0.1)
plot(seq(-6, 6, 0.02), dnorm(seq(-6, 6, 0.02)), type = "l", xlim = c(2.5, 4), ylim = c(0, 
    0.04), col = "grey", ylab = "Y", xlab = "X", lwd = 3, cex.lab = 2, cex.axis = 2)
lines(xx, dt(xx, df = 1), xlim = c(2.5, 4), type = "l", col = "black", lwd = 3)
lines(xx, dt(xx, df = 3), xlim = c(2.5, 4), type = "l", col = "blue", lwd = 3)
lines(xx, dt(xx, df = 9), xlim = c(2.5, 4), type = "l", col = "red", lwd = 3)
lines(xx, dt(xx, df = 45), xlim = c(2.5, 4), type = "l", col = "purple", lwd = 3)
legend(x = 3.5, y = 0.04, legend = c("t1", "t3", "t9", "t45", "Gaussian"), pch = c(20, 
    20, 20, 20, 20), col = c("black", "blue", "red", "purple", "grey"), bty = "n")
title("Tail comparison of t-distribution") 

File: QID-1482-MVAtdistail/MVAtdistail.py
# works on numpy 1.23.5, matplotlib 3.6.2 and scipy 1.10.0
import numpy as np
import matplotlib.pyplot as plt
from scipy.stats import t, norm

xx = np.arange(2.5, 4.1, 0.1)

fig, ax = plt.subplots(1,1,figsize=(10,10))

ax.plot(xx, norm.pdf(xx), 'grey', lw=3, label='Gaussian')
ax.plot(xx, t.pdf(xx, 1), 'black', lw=3, label='t1')
ax.plot(xx, t.pdf(xx, 3), 'blue', lw=3, label='t3')
ax.plot(xx, t.pdf(xx, 9), 'red', lw=3, label='t9')
ax.plot(xx, t.pdf(xx, 45), 'purple', lw=3, label='t45')

ax.set_xlim([2.5, 4])
ax.set_ylim([0, 0.04])
ax.set_xlabel('X', fontsize=20)
ax.set_ylabel('Y', fontsize=20)
ax.set_title('Tail comparison of t-distribution', fontsize=20)
ax.legend(loc='upper right', fontsize=16)

plt.show()
